pipeline/runtime/step_executor.go

```diff
@@ -130,7 +130,7 @@ func (e *StepExecutor) StartStepWithStatusUpdate(ctx context.Context, r *api.Sta
 			e.sendStepStatus(r, &resp)
 			return
 		case <-time.After(defaultStepTimeout):
-			resp = api.VMTaskExecutionResponse{CommandExecutionStatus: api.Failure, ErrorMessage: "step timed out"}
+			resp = api.VMTaskExecutionResponse{CommandExecutionStatus: api.Timeout, ErrorMessage: "step timed out"}
```

Comment:
Want to make sure this doesn't cause a regression.
@raghavharness I remember you added this some time back. Do you know if there's a reason we return failure here (any special handling we do on the platform side?)

Reply:
ah nevermind, I see you revert it back later

Reply:
Yes, I wanted timeout only for the new flow.